src/envoy/http/authn/authenticator_base_test.cc

```diff
@@ -123,6 +127,30 @@ TEST_F(AuthenticatorBaseTest,
 }
 
 // TODO: more tests for Jwt.
+TEST_F(AuthenticatorBaseTest, ValidateJwtWithNoJwtInHeader) {
+  iaapi::Jwt jwt;
+  authenticator_.validateJwt(jwt, [](const Payload* payload, bool success) {
+    // When there is no JWT in the HTTP header, validateJwt() should return
+    // nullptr and failure.
+    EXPECT_TRUE(payload == nullptr);
+    EXPECT_FALSE(success);
+  });
+}
+
+TEST_F(AuthenticatorBaseTest, ValidateJwtWithJwtInHeader) {
+  iaapi::Jwt jwt;
+  Http::TestHeaderMapImpl request_headers_with_jwt{
+      {kSecIstioAuthUserInfoHeaderKey, kSecIstioAuthUserinfoHeaderValue}};
+  FilterContext filter_context{&request_headers_with_jwt, &connection_};
+  MockAuthenticatorBase authenticator{&filter_context};
+
+  authenticator.validateJwt(jwt, [](const Payload* payload, bool success) {
+    // When there is a verified JWT in the HTTP header, validateJwt()
+    // should return non-nullptr and success.
+    EXPECT_TRUE(payload != nullptr);
```

Comment:
it is better to also verify payload data.

Reply:
Done.